sdk/core/azure-core/CHANGELOG.md

```diff
@@ -22,6 +22,10 @@
 - Renamed `TelemetryPolicyOptions` to `TelemetryOptions`.
 - Renamed `ValuePolicyOptions` to `ValueOptions`.
 - Removed `StartTry()` from `Azure::Core::Http::Request`.
+- Changed type of `Azure::Core::Http::RetryOptions::StatusCodes` from `std::vector` to `std::set`.
+- Renamed `Azure::Core::Http::LoggingPolicy` to `LogPolicy`.
+- Introduced `Azure::Core::Http::LogOptions`, a mandatory parameter for `LogPolicy` construction. Entities that are not specified in the allow lists are hidden in the log.
```

Comment:
Because we don't want the clients to accidentally skip initializing the policy. If you are not providing options, it is likely a mistake. At least, if you don't want to have log options in your client, which is unlikely, initialize it with with default explicitly. RetryPolicy doesn't have default parameter either.